pkg/streamingpromql/engine.go

```diff
@@ -44,12 +47,28 @@ func NewEngine(opts EngineOpts, limitsProvider QueryLimitsProvider, metrics *sta
 		return nil, errors.New("enabling delayed name removal not supported by Mimir query engine")
 	}
 
+	// Sort DisabledFunctions to optimise lookups
+	sort.Strings(opts.Features.DisabledFunctions)
+
+	opts.Features.DisabledAggregationsItems = make([]parser.ItemType, 0, len(opts.Features.DisabledAggregations))
+	for _, agg := range opts.Features.DisabledAggregations {
+		item, ok := aggregations.GetAggregationItemType(agg)
+		if !ok {
+			return nil, errors.New(fmt.Sprintf("disabled aggregation '%s' does not exist", agg))
+		}
+		opts.Features.DisabledAggregationsItems = append(opts.Features.DisabledAggregationsItems, item)
+	}
+	// No point sorting DisabledAggregations earlier, as ItemType ints are not in order.
+	sort.Slice(opts.Features.DisabledAggregationsItems, func(i, j int) bool {
+		return opts.Features.DisabledAggregationsItems[i] < opts.Features.DisabledAggregationsItems[j]
+	})
```

Comment:
This should be able to be simplified to `slices.Sort(opts.Features. DisabledAggregationsItems)`.